### PR TITLE
Part 1 of finishing modules object & syntax documentation

### PR DESCRIPTION
**Pull request by GiudGiud**
ticking some boxes mostly but not entirely

### CODE REVIEW

**Comment by lingzou** [comment]
@GiudGiud All looking good. I've made two minor suggestions for your consideration.

**Review by lingzou — COMMENTED** [review]
Two minor comments.

File: modules/navier_stokes/doc/content/source/bcs/INSFEFluidEnergyDirichletBC.md [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
-!! Replace these lines with information regarding the INSFEFluidEnergyDirichletBC object.
-
-## Example Input File Syntax
-
-!! Describe and include an example of how to use the INSFEFluidEnergyDirichletBC object.
+If the [!param](/BCs/INSFEFluidEnergyBC/v_fn) parameter is specified, it is used to compute the boundary fluid
+velocity. If not, the domain velocity variables are used. The velocity is used to determine whether the boundary is
+an inlet.
+
+This boundary condition provides a `conditional' Dirichlet temperature boundary condition for the energy equation, given that no integration by parts is applied to it.
```
Comment (lindsayad) [inline modules/navier_stokes/doc/content/source/bcs/INSFEFluidEnergyDirichletBC.md:11]:
```suggestion
This boundary condition provides a `conditional' Dirichlet temperature boundary condition for the energy equation, given that no integration by parts is applied to the energy advection term.
```

File: modules/navier_stokes/doc/content/source/bcs/INSFEFluidEnergyDirichletBC.md [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
+Without applying integration by parts, the energy equation needs either a Dirichlet temperature boundary condition for an inlet condition,
```
Comment (lindsayad) [inline modules/navier_stokes/doc/content/source/bcs/INSFEFluidEnergyDirichletBC.md:12]:
```suggestion
Without applying integration by parts to the advection term, the energy equation needs a Dirichlet temperature boundary condition for an inlet condition.
```

File: modules/navier_stokes/doc/content/source/bcs/INSFEFluidEnergyDirichletBC.md [hunk 3]
```diff
@@ -1,20 +1,20 @@
...
+or no boundary condition for an outlet condition.
```
Comment (lindsayad) [inline modules/navier_stokes/doc/content/source/bcs/INSFEFluidEnergyDirichletBC.md:13]:
```suggestion
This object additionally introduces a natural boundary condition at outlets.
```

**Review by lindsayad — CHANGES_REQUESTED** [review]

**Review by lindsayad — APPROVED** [review]

**Pull request merged**